internal/controllers/synthetic_devices_controller.go

```diff
@@ -264,33 +271,24 @@ func (vc *SyntheticDevicesController) MintSyntheticDevice(c *fiber.Ctx) error {
 
 	rawPayload := vc.getEIP712Mint(int64(integration.TokenId), vid)
 
-	hash, _, err := signer.TypedDataAndHash(*rawPayload)
+	h, _, err := signer.TypedDataAndHash(*rawPayload)
 	if err != nil {
 		vc.log.Err(err).Msg("Error occurred creating has of payload")
 		return fiber.NewError(fiber.StatusBadRequest, "Couldn't verify signature.")
 	}
+	hash := crypto.Keccak256Hash(h)
+	ownerSignature := common.FromHex(req.OwnerSignature)
 
-	ownerSignature[64] -= 27
-
-	pub, err := crypto.Ecrecover(hash, ownerSignature)
-	if err != nil {
-		vc.log.Err(err).Msg("Error occurred while trying to recover public key from signature")
-		return fiber.NewError(fiber.StatusBadRequest, "Couldn't verify signature.")
-	}
-
-	pubRaw, err := crypto.UnmarshalPubkey(pub)
+	recAddr, err := helpers.Ecrecover(hash, ownerSignature)
 	if err != nil {
-		vc.log.Err(err).Msg("Error occurred marshalling recovered public public key")
-		return fiber.NewError(fiber.StatusBadRequest, "Couldn't verify signature.")
+		vc.log.Err(err).Msg("unable to validate signature")
+		return err
 	}
-
-	payloadVerified := bytes.Equal(crypto.PubkeyToAddress(*pubRaw).Bytes(), userAddr.Bytes())
+	payloadVerified := bytes.Equal(recAddr.Bytes(), userAddr.Bytes())
```

Comment:
You can just check if they're equal!

Reply:
gave `recAddr.Bytes() == userAddr.Bytes()` a try-- looks like recAddr.Bytes() is returning a slice.. did you having something else in mind?

Reply:
`recAddr` is an address, `userAddr` is an address.